### PR TITLE
As a user I should be able to create a new target so I can find people with similar interests in a specific area

### PR DESCRIPTION
**Pull request by keinydev**
[Trello card here](https://trello.com/c/qlE5In1s/3-3-as-a-user-i-should-be-able-to-create-a-new-target-so-i-can-find-people-with-similar-interests-in-a-specific-area)

### CODE REVIEW

File: app/models/target.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Target < ApplicationRecord
+  belongs_to :topic
+  belongs_to :user
+
+  validates_presence_of :title, :radius, :latitude, :longitude, :topic_id
```
Comment (sebastiancaraballo) [inline app/models/target.rb:5]:
```suggestion
  validates :title, :radius, :latitude, :longitude, :topic_id, presence: true
```
Take a look at [this](https://guides.rubyonrails.org/active_record_basics.html#validations).

File: spec/requests/api/v1/targets/create_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  let(:params) do {
+    user: {
+      email: user.email,
+      password: user.password
+    }    
+  }
+  end
+
+  before do
+    user.confirm
```
Comment (sebastiancaraballo) [inline spec/requests/api/v1/targets/create_spec.rb:17]: Instead of doing this, try setting the `confirmed_at` attribute at the user factory.

**Review by sebastiancaraballo — APPROVED** [review]
Nice 💯  Approved after @santir489 comments. Also, try adding code linters (rubocop, reek, etc) to execute on your Github workflow as well.

File: spec/requests/api/v1/targets/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    post new_user_session_path, params: params, as: :json
```
Comment (santir489) [inline spec/requests/api/v1/targets/create_spec.rb:18]: This is going to be used many times in the request specs.
Maybe we can find out another way for doing this?

What about using a spec helper, that defines `authenticated_headers`. And sets the correct headers for a log in user.

So in specific request specs (like this file) we only do something like 
`post api_v1_targets_path, params: target_params, headers: authenticated_headers, as: :json `

Makes sense?

**Review by santir489 — APPROVED** [review]
Great work!

**Pull request closed without merging**